src/InverseKinematicsIPOPT.cpp

```diff
@@ -0,0 +1,499 @@
+#include "InverseKinematicsIPOPT.h"
+
+#include <cassert>
+
+using namespace std;
+using namespace iDynTree;
+using namespace Ipopt;
+
+InverseKinematicsIPOPT::InverseKinematicsIPOPT()
+: exitCode(-12)
+, parentFrame(0)
+, endEffectorFrame(0)
+,modelLoaded(false)
+,framesLoaded(false)
+,gainsLoaded(false)
+,angleError(0)
+{
+    positionError.zero();
+    rotationError.Identity();
+}
+
+InverseKinematicsIPOPT::~InverseKinematicsIPOPT()
+{}
+
+void InverseKinematicsIPOPT::removeJoints(const Model modelInput)
+{
```

Comment:
You are calling a method `removeJoints`, but actually I don't understand what you are removing.